habitat/tasks/rearrange/rearrange_grasp_manager.py

```diff
@@ -5,118 +5,232 @@
 # LICENSE file in the root directory of this source tree.
 
 
+from typing import List, Optional, Tuple
+
 import magnum as mn
 import numpy as np
 
+import habitat.tasks.rearrange.rearrange_sim
+from habitat.config.default import Config
 from habitat.tasks.rearrange.utils import get_aabb
 from habitat_sim.physics import (
     CollisionGroupHelper,
     CollisionGroups,
+    ManagedRigidObject,
     RigidConstraintSettings,
 )
 
 
 class RearrangeGraspManager:
-    def __init__(self, sim, config):
+    """
+    Manages the agent grasping onto rigid objects and the links of articulated objects.
+    """
+
+    def __init__(
+        self,
+        sim: habitat.tasks.rearrange.rearrange_sim.RearrangeSim,
+        config: Config,
+    ) -> None:
+        """Initialize a grasp manager for the simulator instance provided.
+
+        :param config: The task's "SIMULATOR" subconfig node. Defines grasping parameters.
+        """
         self._sim = sim
-        self._snapped_obj_id = None
-        self._snap_constraints = []
-        self._leave_info = None
+        self._snapped_obj_id: Optional[int] = None
+        self._snapped_marker_id: Optional[str] = None
+        self._snap_constraints: List[int] = []
+        self._leave_info: Optional[Tuple[mn.Vector3, float, int]] = None
         self._config = config
 
-    def reset(self):
-        self._leave_info = None
+    def reconfigure(self) -> None:
+        """Removes any existing constraints managed by this structure.
+        Called from _sim.reconfigure().
+        """
+        for constraint_id in self._snap_constraints:
+            self._sim.remove_rigid_constraint(constraint_id)
+        self._snap_constraints.clear()
 
+    def reset(self) -> None:
```

Comment:
Do reconfigure and reset need to be separate? Both called from sim.reconfigure only presently.